src/attendees/attendee.py

```diff
@@ -60,26 +100,114 @@ def has_attribute(self, key: Hashable) -> bool:
         Returns:
             bool: True if Attendee has attribute and vice-versa.
         """
-        raise NotImplementedError
-
+    @property
+    def has_fname(self):    
+        return self._fname == None
+    
+    @property
+    def has_lname(self):    
+        return self._lname == None # already gives a boolean, if statements always accept boolean
+
+    # if condtion, if not condition (i.e. else)
+    
+    @property
+    def has_email(self):    
+        return self._email == None
+
+    @property
+    def has_file_url(self):    
+        return self._file_url == None
+
+    @property
+    def has_file_path(self):    
+        return self._file_url == None
```

Comment:
These `has` methods return the opposite value of what they should. It should return `True` if the attendee has the value and vice-versa. Replace `== None` with `is not None`